### PR TITLE
Return 204 on /ack success instead of JSON true

### PR DESCRIPTION
**Pull request by letmaik**
Addresses first point of https://github.com/microsoft/CCF/issues/1944.

### CODE REVIEW

**Comment by achamayou** [comment]
@letmaik thank you, that looks good. Since this is a REST API change, would you mind adding an entry to CHANGELOG.md?

We could use https://github.com/openapi-tools/open-api-diff, the markdown diff looks quite reasonable, and may even be generated automatically on release.

**Comment by letmaik** [comment]
> 
> 
> @letmaik thank you, that looks good. Since this is a REST API change, would you mind adding an entry to CHANGELOG.md?

Will do, I always forget.

I just realized that CCF only supports `200` when using `set_auto_schema`. Any suggestions on how to extend that?

**Comment by achamayou** [comment]
Pinging @eddyashton, I think it's just a matter of adding a default argument to set_auto_schema() and using it line 337?

**Comment by eddyashton** [comment]
Yes - what @achamayou should be sufficient.

File: tests/infra/member.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def ack(self, remote_node):
         state_digest = self.update_ack_state_digest(remote_node)
         with remote_node.client(*self.auth(write=True)) as mc:
             r = mc.post("/gov/ack", body=state_digest)
-            assert r.status_code == 200, f"Error ACK: {r}"
+            assert r.status_code == 204, f"Error ACK: {r}"
```
Comment (eddyashton) [inline tests/infra/member.py:151]: We have a mix of int literals and `http.HTTPStatus` values in our Python, but I think we should prefer the latter?

```suggestion
            assert r.status_code == http.HTTPStatus.NO_CONTENT, f"Error ACK: {r}"
```

**Comment by letmaik** [comment]
@achamayou @eddyashton I had to do a bit more plumbing but I think I ended up at something good. If `set_auto_schema` is called with a `void` response schema then it defaults to `204` now.

**Comment by achamayou** [comment]
@letmaik that does seem like a sensible default, but we may still want to support user-defined 2xx, like Created etc in the future. Agree it's out of scope for this change though.

**Review by achamayou — APPROVED** [review]

**Comment by letmaik** [comment]
> 
> 
> @letmaik that does seem like a sensible default, but we may still want to support user-defined 2xx, like Created etc in the future. Agree it's out of scope for this change though.

This PR allows specifiying a custom status code. Isn't that what you mean?

**Pull request merged**